### DISCUSSION TITLE
Shared ISR storage and revalidation of multiple instances

### DISCUSSION THREAD

**Discussion by michaelschufi** [vercel/next.js#34517] [Help] [answered]
Hi

I am very excited to see [On-demand Incremental Static Regeneration](https://nextjs.org/blog/next-12-1#on-demand-incremental-static-regeneration-beta) is finally here 🎉.

So now, that this is possible: How can I invalidate the cache of multiple Next.js instances at once? Is there a way I can share the ISR cache between multiple Next.js instances? I use multiple Next.js instances for scaling (SSR/ISR are both used at the moment in a simple docker compose + Traefik setup). 

Am I correct that only an API handler can invalidate the cache? My plan would be to create a simple job that polls a shared storage which holds paths that need to be validated. But this is more challenging if we can only do it in a request context.

**Comment by angeblecon** [comment]
Hi Michael, did you figure out how to handle this situation ? I have the same issue...

**Reply by michaelschufi** [reply]
I have since changed my workplace, so I could not investigate this further. However, what @ijjk posted looks very promising. It not only makes clearing the ISR cache possible, but since it is shared, it comes with the benefit that revalidation only has to happen on one node/pod.

**Comment by ijjk** [comment]
Hi, some related documentation for this and self-hosting has been added here https://nextjs.org/docs/basic-features/data-fetching/incremental-static-regeneration#self-hosting-isr

**Reply by michaelschufi** [reply]
Thank you! That looks very promising. 😃 
Since I cannot verify any more if this solves the issue, I will leave this thread open for some time until we have some feedback from others.

**Reply by shadiramadan** [reply]
Hmm that doesn't specify which specific folders need to share a mount.

**Reply by kesonno** [reply]
Hi @shadiramadan do you figured out what folders need to be shared?

We tried with `.next/cache` but it seems that page cache is not there...

**Reply by shadiramadan** [reply]
@kesonno - Late reply but I just ended up using redis for the page cache.

**Comment by angeblecon** [comment]
Thanks a lot 🙏

**Comment by leerob** [comment, marked as answer]
Hey everyone – we have just published new documentation on self-hosting, including how to configure a custom cache handler for use with ISR / Data Cache in both the Pages and App Router. These docs go in depth about how other features like Image Optimization, Middleware, Environment Variables, Build Caching, and more work when self-hosted. There's also updated Docker based examples.

https://nextjs.org/docs/app/building-your-application/deploying

**Reply by samarpanda** [reply]
Below nextjs doc link specifically talks about configuring the cache in nextjs, behind this we can use any cache i.e Redis etc.
https://nextjs.org/docs/app/building-your-application/deploying#configuring-caching
